### PR TITLE
Change closed to a real state

### PR DESCRIPTION
**Pull request by pixeltrix**
As part of a series of steps to simplify our list queries make the closed state a real state and not based on it being open and the closed_at time being in the past.

### CODE REVIEW

**Comment by mybits** [comment]
:+1: for simplifying things

**Comment by h-lame** [comment]
``Petition#open?`` is no longer symmetrical with ``Petition#closed?`` depending on the time you run them.  Something can be ``open? #=> true`` and ``closed? #=> true`` if the in-memory instance crossed the ``closed_at`` boundary.  It's minor, sure, but it does mean we've got some danger in how we write our views and controller actions if we want to guard against folk loading a page just before midnight and submitting a signature just after.

There is ``Petition#can_be_signed?`` which has the logic we want, maybe we should merge that with ``#open?``.  Or get rid of it.

Separately, but related, our scopes for fetching open and closed petitions don't care about ``closed_at`` but our instance methods do.  It's fine (we've done it for a performance reasons) but we maybe want to leave a note somewhere in the code.

File: app/models/petition.rb [hunk 1]
```diff
@@ -229,8 +231,8 @@ def hidden?
     state == HIDDEN_STATE
   end
 
-  def closed?
-    state == OPEN_STATE && closed_at <= Time.current
+  def closed?(time = Time.current)
```
Comment (h-lame) [inline app/models/petition.rb]: Total nitpick: but we don't need this arg anymore.

File: lib/tasks/import.rake [hunk 2]
```diff
@@ -35,7 +35,7 @@ namespace :import do
         petition.title              = row['title']
         petition.description        = row['description']
         petition.response           = row['response']
-        petition.state              = row['state']
+        petition.state              = row['state'] == 'open' ? 'closed' : row['state']
```
Comment (h-lame) [inline lib/tasks/import.rake]: This is just brilliant, not confusing at all ;)

File: spec/models/petition_spec.rb [hunk 3]
```diff
@@ -474,117 +468,173 @@
...
+        expect(petition.closed?).to be_truthy
+      end
+    end
+
+    context "when the closed_at has not passed but the state is closed" do
+      let(:petition) { FactoryGirl.build(:petition, state: Petition::CLOSED_STATE, closed_at: 3.days.from_now) }
+
+      it "return true" do
+        expect(petition.closed?).to be_truthy
+      end
```
Comment (h-lame) [inline spec/models/petition_spec.rb]: Can we just collapse these down to "is closed when state is closed"? like the others?  I admit that I find the title of `"when the closed_at has passed but the state is still open"` a bit confusing (compared to the implementation which uses `CLOSED_STATE`, not `OPEN_STATE`.

**Pull request merged**